vars/githubChangeset.groovy

```diff
@@ -0,0 +1,26 @@
+// vars/githubChangeset.groovy
+// Iterates over commits since the last build and outputs them 
+// in the following format via echo:
+//    - Commit message [author]
+
+def call() {
+	def changeString = ""
+ 	def changeLogSets = currentBuild.changeSets
+ 	
+ 	// Iterate over changes and add them to changeString
+ 	for (int i = 0; i < changeLogSets.size(); i++) {
+		def entries = changeLogSets[i].items
+ 		for (int j = 0; j < entries.length; j++) {
+ 			def entry = entries[j]
+ 			changeString += " - ${entry.msg} [${entry.author}]\n"
```

Comment:
Timestamp would be nice per https://support.cloudbees.com/hc/en-us/articles/217630098-How-to-access-Changelogs-in-a-Pipeline-Job- 

Also, it would be awesome to do this with pure shell, but I guess the hard part is figuring out the last built commit - `git log --graph --pretty=format:'%Cred%h%Creset -%C(yellow)%d%Creset %s %Cgreen(%cr) %C(bold blue)<%an>%Creset' --abbrev-commit 68e07c5`